src/main/java/at/hannibal2/skyhanni/features/slayer/SlayerMiniBossFeatures.kt

```diff
@@ -1,83 +1,58 @@
 package at.hannibal2.skyhanni.features.slayer
 
 import at.hannibal2.skyhanni.SkyHanniMod
-import at.hannibal2.skyhanni.events.EntityMaxHealthUpdateEvent
+import at.hannibal2.skyhanni.data.SlayerAPI
+import at.hannibal2.skyhanni.data.mob.Mob
 import at.hannibal2.skyhanni.events.LorenzRenderWorldEvent
-import at.hannibal2.skyhanni.events.LorenzWorldChangeEvent
-import at.hannibal2.skyhanni.features.combat.damageindicator.DamageIndicatorManager
-import at.hannibal2.skyhanni.features.dungeon.DungeonAPI
-import at.hannibal2.skyhanni.mixins.hooks.RenderLivingEntityHelper
+import at.hannibal2.skyhanni.events.MobEvent
 import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
-import at.hannibal2.skyhanni.utils.CollectionUtils.editCopy
-import at.hannibal2.skyhanni.utils.ColorUtils.withAlpha
 import at.hannibal2.skyhanni.utils.EntityUtils.canBeSeen
-import at.hannibal2.skyhanni.utils.EntityUtils.hasMaxHealth
 import at.hannibal2.skyhanni.utils.LorenzColor
-import at.hannibal2.skyhanni.utils.LorenzUtils
 import at.hannibal2.skyhanni.utils.RenderUtils.drawLineToEye
 import at.hannibal2.skyhanni.utils.getLorenzVec
-import net.minecraft.entity.EntityCreature
-import net.minecraft.entity.monster.EntityBlaze
-import net.minecraft.entity.monster.EntityEnderman
-import net.minecraft.entity.monster.EntitySpider
-import net.minecraft.entity.monster.EntityZombie
-import net.minecraft.entity.passive.EntityWolf
 import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
 
 @SkyHanniModule
 object SlayerMiniBossFeatures {
 
     private val config get() = SkyHanniMod.feature.slayer
-    private var miniBosses = listOf<EntityCreature>()
+    private var miniBosses = mutableSetOf<Mob>()
 
     @SubscribeEvent
-    fun onEntityHealthUpdate(event: EntityMaxHealthUpdateEvent) {
-        if (!isEnabled()) return
-        val entity = event.entity as? EntityCreature ?: return
-        if (DamageIndicatorManager.isBoss(entity)) return
-
-        val maxHealth = event.maxHealth
-        for (bossType in SlayerMiniBossType.entries) {
-            if (!bossType.health.any { entity.hasMaxHealth(it, true, maxHealth) }) continue
-            if (bossType.clazz != entity.javaClass) continue
-
-            miniBosses = miniBosses.editCopy { add(entity) }
-            RenderLivingEntityHelper.setEntityColorWithNoHurtTime(
-                entity,
-                LorenzColor.AQUA.toColor().withAlpha(127)
-            ) { config.slayerMinibossHighlight }
-        }
+    fun onMobSpawn(event: MobEvent.Spawn.SkyblockMob) {
+        if (!SlayerAPI.isInAnyArea) return
+        val miniBossType = SlayerAPI.getSlayerTypeForCurrentArea()?.miniBossType ?: return
```

Comment:
this approach will still fail while the player stands outside a slayer area and the mini boss spawns couple blocks away inside the area. please remove the "valid current area" check entirely and instad check if the name of the spawned mob matches any mob name from any slayer type